doc/classes/Input.xml

```diff
@@ -30,6 +30,68 @@
 				If the specified action is already pressed, this will release it.
 			</description>
 		</method>
+		<method name="add_arbitrary_joy">
+			<return type="int" />
+			<param index="0" name="name" type="String" />
+			<description>
+				Returns the device index of a newly created arbitrary joypad.
+				An arbitrary joypad allows data to be sent directly to the input system, which will then be parsed as native input data through [method set_arbitrary_joy_axis] and [method set_arbitrary_joy_button]. This can be useful if you have a means of acquiring data from a specific device that isn't natively recognized by Godot's input system.
+				[b]Note:[/b] An arbitrary joypad will not be removed automatically, you must call [method remove_arbitrary_joy] manually.
+				[b]Example:[/b]
+				[codeblocks]
+				[gdscript]
+				var _index := -1
+
+				func create_joy(name := ""):
+				    if _index == -1:
+				        _index = Input.add_arbitrary_joy(name)
+
+				func process_inputs(bytes: PackedByteArray):
+				    if _index != -1:
+				        Input.set_arbitrary_joy_axis(_index, JOY_AXIS_LEFT_X, bytes[0])
+				        Input.set_arbitrary_joy_axis(_index, JOY_AXIS_LEFT_Y, bytes[1])
+				        Input.set_arbitrary_joy_button(_index, JOY_BUTTON_A, bytes[2] &gt; 0)
+				        Input.set_arbitrary_joy_button(_index, JOY_BUTTON_B, bytes[3] &gt; 0)
+
+				func destroy_joy():
+				    if _index != -1:
+				        Input.remove_arbitrary_joy(_index)
+				        _index = -1
+				[/gdscript]
+				[csharp]
+				private int _index = -1;
+
+				public CreateJoy(string name = "")
```

Comment:
I'm assuming you meant for this to be a void returning method.

```suggestion
				public void CreateJoy(string name = "")
```

Reply:
Forgot to add that after changing it from a constructor, good catch!